packages/ui/lib/RadioGroup/index.tsx

```diff
@@ -74,71 +30,105 @@ function useBaseRadioGroup({
   onChange,
   children,
 }: UseBaseRadioGroupProps) {
-  return {
-    // TODO: Should probably have the groupElem return a wrapperDiv so that it would
-    // be easy to for example change the Radios into a horizontal layout? Or maybe -
-    // this should be left to the consumer?
-    groupElem: (
-      <BaseRadioGroupContext.Provider value={{ activeValue: value, onChange }}>
-        {children}
-      </BaseRadioGroupContext.Provider>
-    ),
-  };
+  return (
+    <BaseRadioGroupContext.Provider value={{ activeValue: value, onChange }}>
+      {children}
+    </BaseRadioGroupContext.Provider>
+  );
 }
 
-export const Group = (props: UseBaseRadioGroupProps) => {
-  const { groupElem } = useBaseRadioGroup(props);
-  return groupElem;
+export const Group = ({
+  id,
+  className,
+  style,
+  ariaLabeledBy,
+  ...rest
+}: UseBaseRadioGroupProps & {
+  id?: string;
+  className?: string;
+  style?: CSSProperties;
+  ariaLabeledBy?: string;
+}) => {
+  const groupElem = useBaseRadioGroup(rest);
+
+  return (
+    <div
+      aria-labelledby={ariaLabeledBy}
+      className={cn(className, styles.group)}
+      id={id}
+      role="radiogroup"
+      style={style}
+    >
+      {groupElem}
+    </div>
+  );
 };
 
 // ------------------------------------------------------------------------------
 // 👉 ITEM
 // ------------------------------------------------------------------------------
 
-type RadioGroupItemProps = {
-  value: string;
+type UseBaseRadioGroupItemProps = {
+  id?: string;
+  className?: string;
+  style?: CSSProperties;
   label: string;
+  name?: string;
+  title?: string;
+  value: string;
+  ariaLabelledBy?: string;
+  ariaDescribedBy?: string;
 };
 
-// QUESTION: In this case, we do not need to merge any label
-// props as we are not exposing it to the user?
-export type BaseRadioGroupItemProps = Sys42Props<
-  RadioGroupItemProps,
-  Partial<
-    Omit<InputHTMLAttributes<HTMLInputElement>, "checked" | "readOnly" | "type">
-  >
->;
-
-function Item({
-  value,
-  label,
-  onClick,
-  className,
-  ...nativeProps
-}: BaseRadioGroupItemProps) {
+function useBaseItem(props: UseBaseRadioGroupItemProps) {
   const { activeValue, onChange } = useContext(BaseRadioGroupContext);
 
-  function handleClick(e: React.MouseEvent<HTMLInputElement>) {
-    onClick?.(e);
-    onChange(value);
+  function handleChange() {
+    onChange(props.value);
   }
 
+  return {
+    handleChange,
+    checked: props.value === activeValue,
+    ...omit(props, "value"),
+  };
+}
+
+function Item(props: UseBaseRadioGroupItemProps) {
+  const {
+    label,
+    checked,
+    handleChange,
+    className,
+    name,
+    title,
+    style,
+    id,
+    ariaDescribedBy,
+    ariaLabelledBy,
+  } = useBaseItem(props);
```

Comment:
I think this hook should return labelProps, labelRef, inputProps, inputRef and isChecked